bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/watch/JRubyGemWatchService.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrubyscripting.internal.watch;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.util.Arrays;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.service.AbstractWatchService;
+
+/**
+ * Watches a gem home
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+@NonNullByDefault
+public class JRubyGemWatchService extends AbstractWatchService {
```

Comment:
I'm not so happy with these tons of new watch services, but this might improve after we have https://github.com/openhab/openhab-core/pull/3004 merged which simplifies the interface.

Reply:
Oooh. That'll be nice. To slightly assuage your concerns in the meantime, while this is setup to allow an arbitrary number of watch services, 99% of the time it will be the default config, and there will be just 2 - one for GEM_HOME, and a single one for RUBYLIB. And both of those will be directories nested underneath the JRubyScriptFileWatcher, so the underlying WatchQueueReader won't actually be watching any more, but will have more bookkeeping.